make/libraries

```diff
@@ -15,20 +15,24 @@ CPPLINT ?= $(MATH)lib/cpplint_4.45
 ##
 
 SUNDIALS_CVODES := $(patsubst %.c,%.o,\
-  $(addprefix $(SUNDIALS)/src/cvodes/, cvodes.c cvodes_io.c cvodea.c cvodea_io.c cvodes_direct.c cvodes_diag.c cvodes_spils.c cvodes_bandpre.c cvodes_bbdpre.c) \
+  $(wildcard $(SUNDIALS)/src/cvodes/*.c) \
   $(wildcard $(SUNDIALS)/src/sundials/*.c) \
-  $(wildcard $(SUNDIALS)/src/sunmat_band/*.c) \
-  $(wildcard $(SUNDIALS)/src/sunmat_dense/*.c) \
-  $(wildcard $(SUNDIALS)/src/sunlinsol_band/*.c) \
-  $(wildcard $(SUNDIALS)/src/sunlinsol_dense/*.c))
+  $(wildcard $(SUNDIALS)/src/sunmat_band/[^f]*.c) \
```

Comment:
Question: What does the `[^f]` do?

I didn't expect these changes and I haven't been able to find a reference that explains what that is. Also, should line 18 also have it?

Reply:
The f* files are used by sundials to provide fortran compatible interfaces somehow. We don't need those - it would be a waste of compile time. Line 18 & 19 don't need this as there are no files starting with f.

Reply:
Awesome. I think it warrants writing down. Especially since we're not in charge of Sundials and this convention isn't anything they guarantee.